$DragonFly: src/gnu/usr.bin/groff/src/libs/libgroff/patches/cmap.cpp.patch,v 1.1 2005/09/01 20:47:13 asmodai Exp $

--- cmap.cpp	2005-08-31 14:28:07.000000000 +0200
+++ cmap.cpp	2005-08-31 15:15:08.000000000 +0200
@@ -19,12 +19,15 @@
 along with this program. If not, see <http://www.gnu.org/licenses/>. */
 
 #include <ctype.h>
+#if defined(__DragonFly__) || defined(__FreeBSD__)
+#include <locale.h>
+#endif
 #include "cmap.h"
 
 cmap cmlower(CMAP_BUILTIN);
 cmap cmupper(CMAP_BUILTIN);
 
-#ifdef isascii
+#if defined(isascii) && (!defined(__DragonFly__) || !defined(__FreeBSD__))
 #define ISASCII(c) isascii(c)
 #else
 #define ISASCII(c) (1)
@@ -49,6 +52,9 @@
   if (initialised)
     return;
   initialised = 1;
+#if defined(__DragonFly__) || (__FreeBSD__)
+  (void)setlocale(LC_CTYPE, "");
+#endif
   for (int i = 0; i <= UCHAR_MAX; i++) {
     cmupper.v[i] = ISASCII(i) && islower(i) ? toupper(i) : i;
     cmlower.v[i] = ISASCII(i) && isupper(i) ? tolower(i) : i;
